### PR TITLE
[charts] Take responsive container from data grid

### LINKED ISSUE #9470: [charts] Improve responsive charts container
The responsive container is roughly taken from the hackathon. Might be better to have a look at the data grid since it benefits from 2 years of feedback

### PR DESCRIPTION
**Pull request by alexfauquette**
Fix #9470

### CODE REVIEW

**Review by flaviendelangle — APPROVED** [review]

File: packages/x-charts/src/ResponsiveChartContainer/index.tsx [hunk 1]
```diff
@@ -1,53 +1,120 @@
...
+    const newWidth = parseFloat(computedStyle.width) || 0;
+
+    setWidth(newWidth);
+    setHeight(newHeight);
+  }, []);
+
   React.useEffect(() => {
-    const element = ref.current;
-    if (element === null || (inHeight !== undefined && inWidth !== undefined)) {
+    // Ensure the error detection occures after first rendering
```
Comment (LukasTy) [inline packages/x-charts/src/ResponsiveChartContainer/index.tsx:36]:
```suggestion
    // Ensure the error detection occurs after the first rendering
```

File: packages/x-charts/src/ResponsiveChartContainer/index.tsx [hunk 2]
```diff
@@ -1,53 +1,120 @@
...
   inHeight?: number,
 ): [React.RefObject<HTMLDivElement>, number, number] => {
-  const ref = React.useRef<HTMLDivElement>(null);
+  const rootRef = React.useRef<HTMLDivElement>(null);
+  const displayError = React.useRef<boolean>(false);
 
   const [width, setWidth] = React.useState(0);
   const [height, setHeight] = React.useState(0);
 
+  const computeSizeAndPublishResizeEvent = React.useCallback(() => {
```
Comment (LukasTy) [inline packages/x-charts/src/ResponsiveChartContainer/index.tsx:18]: Am I missing something or is the function name somewhat misleading?
I don't see any `publishing` of a `resizeEvent`... 🙈

**Review by LukasTy — APPROVED** [review]

File: packages/x-charts/src/ResponsiveChartContainer/index.tsx (hunk 2 above)
Comment (flaviendelangle) [reply packages/x-charts/src/ResponsiveChartContainer/index.tsx:18]: It's the grid name which is coherent, but I agree here it does not make sense

Comment (alexfauquette) [reply packages/x-charts/src/ResponsiveChartContainer/index.tsx:18]: Effectively, I was too focused on the content I forget the naming 🙈

File: packages/x-charts/package.json [hunk 3]
```diff
@@ -39,7 +39,6 @@
   },
   "dependencies": {
     "@babel/runtime": "^7.22.5",
-    "@juggle/resize-observer": "^3.4.0",
```
Comment (oliviertassinari) [inline packages/x-charts/package.json:42]: 3 kB saved https://bundlephobia.com/package/@juggle/resize-observer@3.4.0 🚀

**Pull request merged**